gobblin-modules/gobblin-compliance/src/main/java/gobblin/compliance/restore/ComplianceRestoreJob.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package gobblin.compliance.restore;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CountDownLatch;
+
+import org.apache.commons.lang.exception.ExceptionUtils;
+import org.apache.thrift.TException;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.util.concurrent.FutureCallback;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+
+import javax.annotation.Nullable;
+import lombok.extern.slf4j.Slf4j;
+
+import gobblin.compliance.ComplianceConfigurationKeys;
+import gobblin.compliance.ComplianceEvents;
+import gobblin.compliance.ComplianceJob;
+import gobblin.compliance.utils.ProxyUtils;
+import gobblin.configuration.State;
+import gobblin.dataset.Dataset;
+import gobblin.dataset.DatasetsFinder;
+import gobblin.util.ExecutorsUtils;
+import gobblin.util.reflection.GobblinConstructorUtils;
+
+
+/**
+ * Job for restoring {@link gobblin.compliance.HivePartitionDataset} from the backups
+ *
+ * @author adsharma
+ */
+@Slf4j
+public class ComplianceRestoreJob extends ComplianceJob {
```

Comment:
This looks almost the same as ComplianceRetentionJob. Would it make sense to share the code through another class?